src/Form/matching.js

```diff
@@ -299,6 +300,8 @@ class Matching {
                     return undefined
                 }
             }
+
+            if (result?.skip) break
```

Comment:
This is the main logic change. Now when we set the `skip` property to `true`, we break the entire type, rather than skipping to the next strategy. This is probably the initial intent of this feature and was probably just a bug.